GH-2 New script structure

viniciusccarvalho opened the pull request:
This is thew new layout to deploy, run tests and clean up for cloudfoundry and local servers.

Please take a look at the readme on scripts/ for instructions

cppwfs commented on scripts/local/common.sh:
```diff
@@ -0,0 +1,68 @@
+    exit 1
+  fi
+
+<<<<<<< HEAD
```
Left over from a merge

cppwfs commented on scripts/README.adoc:
```diff
@@ -0,0 +1,108 @@
+= Spring Cloud Data Flow Acceptance Tests =
+
+This project bootstrap a dataflow server on a target platform, executes a series of tests by deploying some
```
replace "deploying some apps" with, "creating a series of streams and tasks"

cppwfs commented:
May want to add notes on being sure to setup JAVA_HOME and DOCKER_HOST.
After starting a local Rabbit instance and MySQL and setting a JAVA_HOME I got the following exception for local:
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:58)
Caused by: java.lang.IllegalArgumentException: Invalid argument syntax: --spring.cloud.dataflow.applicationProperties.stream.spring.rabbitmq.host=
	at org.springframework.core.env.SimpleCommandLineArgsParser.parse(SimpleCommandLineArgsParser.java:75)
	at org.springframework.core.env.SimpleCommandLinePropertySource.<init>(SimpleCommandLinePropertySource.java:87)
	at org.springframework.boot.DefaultApplicationArguments$Source.<init>(DefaultApplicationArguments.java:76)
	at org.springframework.boot.DefaultApplicationArguments.<init>(DefaultApplicationArguments.java:42)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:306)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1186)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1175)
	at org.springframework.cloud.dataflow.server.local.LocalDataFlowServer.main(LocalDataFlowServer.java:35)

viniciusccarvalho commented:
@cppwfs Thanks for the review, fixed the spell checking. There's already instructions to set DOCKER_HOST on the pre-requisites of the local deployment. Added JAVA_HOME as well.

viniciusccarvalho commented:
@cppwfs Found the issue on the local mysql, need to fix that bug.

cppwfs commented on scripts/run.sh:
```diff
@@ -0,0 +1,171 @@
+      run_scripts $BINDER "create.sh"
+    popd
+    run_scripts "redis" "create.sh"
+    run_scripts "server" "create.sh"
```
May want to add a health check to make sure Dataflow is up and running prior to test execution.

i.e. https://github.com/spring-cloud/spring-cloud-dataflow-acceptance-tests/blob/master/startLocalAcceptance.sh#L330

cppwfs commented on scripts/local/common.sh:
```diff
@@ -0,0 +1,90 @@
+
+function stop_docker_container() {
+
+  [ -z "$DOCKER_HOST" ] && { echo "Environment variable DOCKER_HOST must be set"; exit 1; }
```
Having issues setting DOCKER_HOST on my local mac using the Docker Toolbox.  According to instructions it should not be set: https://docs.docker.com/docker-for-mac/docker-toolbox/#setting-up-to-run-docker-for-mac.   Also we did not need to set the DOCKER_HOST for docker-compose when running the tests on the Jenkins CI machine.  So for now the docker commands fail to fire when following instructions in README :-(

cppwfs replied:
Resolved by running `docker-machine regenerate-certs`

cppwfs commented on scripts/run.sh:
```diff
@@ -0,0 +1,171 @@
+if [ -z "$skipTests" ]; then
+  run_tests
+fi
+if [ -z "$skipCleanup" ]; then
```
If we support a skipCleanup phase.  We also need to have an option that cleans up previously setup environment.

viniciusccarvalho replied:
Just run with skip setup and it will clean up

cppwfs replied:
gotcha

cppwfs commented on scripts/run.sh:
```diff
@@ -0,0 +1,171 @@
+}
+
+function download(){
+  if [ ! -f "$1/scdf-server.jar" ]; then
```
For the local user:  We probably want to have it pull a fresh copy of the jar each time.  This is so that they won't get a false positive if pulling from snapshots that can change many times during the day.  -or- have an option so that they can use the existing copy vs. downloading a fresh one.

viniciusccarvalho commented on scripts/local/mysql/create.sh:
```diff
@@ -0,0 +1,7 @@
+
+create "mysql" 3306
+
+APPLICATION_ARGS="$APPLICATION_ARGS  --spring.datasource.url=jdbc:mysql://$SERVICE_HOST/scdf --spring.datasource.username=spring --spring.datasource.password=cloud  --spring.datasource.driver-class-name=org.mariadb.jdbc.Driver"
```
Just added that capability as exported variables
